### PR TITLE
SourceKittenDaemon support

### PR DESCRIPTION
**Pull request by smallfx**
Probably needs changes before being merged (maybe the sourcekitten fallback support should happen first?)

### CODE REVIEW

**Comment by landaire** [comment]
Looks pretty good. Sorry for the delay in getting back on this, I was out and about for most of the weekend.

If I'm not mistaken, deoplete loads the plugin per vim session, and when the completion plugin is invoked for completion it'll have the same state (minus context) no matter what file/project triggered the completion. Is that correct? Going off that, I think that we can make autoloading work by:
1. Create a dictionary of project directories and the port sourcekitten is listening on
2. When the completion is invoked, check if any key in the dictionary matches the start of the path for the file.
   1. If no project is found, traverse the filesystem up until an Xcode project is hit
      - If no Xcode project is found, invoke sourcekitten instead and quit here
   2. Start sourcekittendaemon on the most recently used port, + 1 (starting from MRU being 8080)
   3. Add to dictionary
3. Make a request to sourcekittendaemon for the given project's port.

@smallfx @brendanjerwin I've added both of you as collaborators to this project. Nothing is expected, I just figured that it would be easier to collaborate on features like this with a separate branch on the same repo.

File: rplugin/python3/deoplete/sources/swift.py [hunk 1]
```diff
@@ -18,40 +18,38 @@ def __init__(self, vim):
...
+            os.makedirs(self.temp_file_directory)
 
     def get_complete_position(self, context):
         m = re.search(r'\w*$', context['input'])
         return m.start() if m else -1
 
     def gather_candidates(self, context):
         line = self.vim.current.window.cursor[0]
-        column = context['complete_position']
+        column = self.vim.current.window.cursor[1]
```
Comment (brendanjerwin) [inline rplugin/python3/deoplete/sources/swift.py:33]: Why this change?

**Comment by smallfx** [comment]
@landaire - i'm going to close this PR (which i half expected to do anyway) in favor of @mitsuse's pull request #8 

it has changes that neatly set the groundwork for adding SourceKittenDaemon while also still having SourceKitten as a "fallback".

i'm gonna merge #8 and then i'm gonna write the SourceKittenDaemon stuff on top of that!
